shared/search/user-search/search-bar.desktop.js

```diff
@@ -1,53 +1,49 @@
 // @flow
 import React, {Component} from 'react'
 import _ from 'lodash'
-
 import {Box, Icon, Input, Text} from '../../common-adapters'
 import {globalStyles, globalColors, transition} from '../../styles'
 import {platformToLogo24} from '../../constants/search'
+import {withState, compose, withHandlers} from 'recompose'
 
 import type {SearchPlatforms} from '../../constants/search'
 import type {ServiceFn} from './render'
 import type {IconType} from '../../common-adapters/icon'
 import type {Props} from './search-bar'
 
-type ServiceIconState = {showingTooltip: boolean}
-type ServiceIconProps = {serviceName: SearchPlatforms, tooltip: string,
-  iconType: IconType, selected: boolean, onClickService: ServiceFn}
-class ServiceIcon extends Component<void, ServiceIconProps, ServiceIconState> {
-  state: ServiceIconState;
-
-  constructor (props: ServiceIconProps) {
-    super(props)
-
-    this.state = {
-      showingTooltip: false,
-    }
-  }
-
-  render () {
-    const {serviceName, tooltip, iconType, selected, onClickService} = this.props
-    return (
-      <Box style={{...serviceContainerStyle, backgroundColor: selected ? globalColors.blue4 : null}}
-        onMouseEnter={() => this.setState({showingTooltip: true})}
-        onMouseLeave={() => this.setState({showingTooltip: false})}
-        onClick={() => onClickService(serviceName)} >
-        <Icon type={iconType} style={{...serviceIconStyle,
-          opacity: selected || this.state.showingTooltip ? 1.0 : 0.6}} />
-        <Text type='BodySmall' style={{...serviceTooltipStyle,
-          opacity: this.state.showingTooltip ? 1 : 0}}>{tooltip}</Text>
-      </Box>
-    )
-  }
+type ServiceIconProps = {
+  serviceName: SearchPlatforms,
+  tooltip: string,
+  iconType: IconType,
+  selected: boolean,
+  onClickService: ServiceFn,
+  onMouseEnter: () => void,
+  onMouseLeave: () => void,
+  onClick: () => void,
+  showingTooltip: boolean,
 }
 
-class SearchBar extends Component<void, Props, void> {
-  _onDebouncedSearch: (overridePlatform?: SearchPlatforms) => void;
+const ServiceIcon =
+  compose(
+    withState('showingTooltip', 'setShowingTooltip'),
+    withHandlers({
+      onClick: props => event => props.onClickService(props.serviceName),
+      onMouseEnter: props => event => props.setShowingTooltip(true),
+      onMouseLeave: props => event => props.setShowingTooltip(false),
+    })
+  )(({serviceName, tooltip, iconType, selected, onClick, onMouseEnter, onMouseLeave, showingTooltip}: ServiceIconProps) => (
```

Comment:
:sheep: I think the render function should be broken out of this big expression. It's getting a bit unwieldy w/ `)(({`.